app/filetransfer/sftp.go

```diff
@@ -0,0 +1,79 @@
+package filetransfer
+
+import (
+	"encoding/csv"
+	"fmt"
+	"net"
+	"path/filepath"
+
+	"github.com/companieshouse/chs.go/log"
+	"github.com/companieshouse/payment-reconciler/app/config"
+	"github.com/companieshouse/payment-reconciler/app/models"
+	"github.com/pkg/sftp"
+	"golang.org/x/crypto/ssh"
+)
+
+// SFTP provides a concrete implementation of the FileTransfer interface, transferring files to an SFTP server
+type SFTP struct {
+	Config          *config.Config
+	SSHClientConfig *ssh.ClientConfig
+}
+
+// New returns a new SFTP struct using the provided config
+func New(cfg *config.Config) *SFTP {
+
+	sshCfg := &ssh.ClientConfig{
+		User: cfg.SFTPUserName,
+		HostKeyCallback: func(hostname string, remote net.Addr, key ssh.PublicKey) error {
+			return nil
+		},
+		Auth: []ssh.AuthMethod{
+			ssh.Password(cfg.SFTPPassword),
+		},
+	}
+
+	sshCfg.SetDefaults()
+
+	return &SFTP{
+		Config:          cfg,
+		SSHClientConfig: sshCfg,
+	}
+}
+
+// UploadCSVFiles uploads an array of CSV's to an STFP server
+func (t *SFTP) UploadCSVFiles(csvs []models.CSV) error {
+
+	log.Info("Starting upload of CSV's. Initiating SSH connection to " + t.Config.SFTPServer)
+
+	client, err := ssh.Dial("tcp", t.Config.SFTPServer+":22", t.SSHClientConfig)
```

Comment:
Could we add SFTP port in global_env . Other applications could reuse it If needed.

Reply:
Port 22 is the standard SSH port, it doesn't need to be configurable since this will never change.

Reply:
Port made configurable